compiler/parser/src/token.rs

```diff
@@ -240,6 +240,23 @@ pub enum StringKind {
     Unicode,
 }
 
+impl TryFrom<String> for StringKind {
```

Comment:
instead of requiring an allocation here, why not `TryFrom` for `char` and `[char; 2]`? This is slightly more code but should avoid the `to_string` calls in `lex_identifier`.

Reply:
@youknowone any better ideas?

Reply:
> why not TryFrom for char and [char; 2]?

Sounds good. I'll make that change.